### DISCUSSION TITLE
Will axum_login work with webauthn

### DISCUSSION THREAD

**Discussion by Kuzat** [maxcountryman/axum-login#204] [Q&A]
How would we use [webauthn-rs](https://github.com/kanidm/webauthn-rs) with axum_login to authenticate the user with webauthn?

**Comment by maxcountryman** [comment]
I haven't looked over that specific crate, but I am aware of different implementation that could be adapted.

However, that said, I believe there's some need to leverage the browser web APIs to complete the workflow (so it might require JavaScript on the client side of things).

Anyway I'm sorry I don't immediately have an example for you but I'd be interested in adding one if someone wanted to take that on: the question we'd need to answer is if webauthn will work within the API of `authenticate`. (It should, but I don't know what machinery around that is required.)

**Reply by Kuzat** [reply]
Thanks for the quick answer! I am experimenting with it at the moment and will report back if I figure out how to do it.

**Reply by mainrs** [reply]
@Kuzat did you manage to figure it out? :)
